src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonOutputFormatter.cs

```diff
@@ -61,6 +75,51 @@ public override async Task WriteResponseBodyAsync(OutputFormatterWriteContext co
             context.HttpContext.AllowSynchronousIO();
 
             HttpResponse response = context.HttpContext.Response;
+
+            var elementsSearchParameter = context.HttpContext.GetElementsSearchParameter(_logger);
+
+            Resource resource = null;
+
+            if (context.Object is Hl7.Fhir.Model.Bundle)
+            {
+                var bundle = context.Object as Hl7.Fhir.Model.Bundle;
+                resource = bundle;
+
+                if (elementsSearchParameter != null && elementsSearchParameter.Any())
+                {
+                    // _elements is not supported for a raw resource, revert to using FhirJsonSerializer
+                    foreach (var rawBundleEntryComponent in bundle.Entry)
+                    {
+                        if (rawBundleEntryComponent is RawBundleEntryComponent)
+                        {
+                            rawBundleEntryComponent.Resource = ((RawBundleEntryComponent)rawBundleEntryComponent).ResourceElement.ToPoco<Resource>(_deserializer);
+                        }
+                    }
+                }
+                else if (bundle.Entry.All(x => x is RawBundleEntryComponent))
+                {
+                    await _bundleSerializer.Serialize(context.Object as Hl7.Fhir.Model.Bundle, context.HttpContext.Response.Body);
+                    return;
+                }
+            }
+            else if (context.Object is RawResourceElement)
```

Comment:
It looks like the output stream optimization does not kick in for POST/PUT.

Reply:
Yeah I haven't changed any of the write handlers to return using the raw resource code path. I was planning to do it as a separate work item.

Reply:
Sounds good